nixos/modules/services/networking/prosody.nix

```diff
@@ -596,7 +601,17 @@ let
         ${lib.optionalString (cfg.httpFileShare.http_host != null) ''
           http_host = "${cfg.httpFileShare.http_host}"
         ''}
-        ${settingsToLua "  http_file_share_" (cfg.httpFileShare // { domain = null; })}
+        ${lib.optionalString (cfg.httpFileShare.http_external_url != null) ''
+          http_external_url = "${cfg.httpFileShare.http_external_url}"
+        ''}
```

Comment:
Since this is a freeform setting, those 3 lines should not be necessary or is ordering important?

Reply:
Ordering is not important, but there shouldn't be `http_file_share_` which is added to all freeform settings.